### ISSUE TITLE
Image estimator fails with reflected regions

### ISSUE THREAD

**Issue by mackaiver** [gammapy/gammapy#1252]
Running
```
bkg_estimator = ReflectedRegionsBackgroundEstimator(
    obs_list=obs_list,
    on_region=on_region,
    exclusion_mask=exclusion_mask,
)
bkg_estimator.run()

image_estimator = IACTBasicImageEstimator(
    reference=ref_image,
    emin=100 * u.GeV,
    emax=100 * u.TeV,
    offset_max=3 * u.deg,
    background_estimator=bkg_estimator,
    exclusion_mask=exclusion_mask,
)
image_estimator.run(obs_list)
```

yields:

```
TypeError                                 Traceback (most recent call last)
~/Development/galactic_plane_cta/skymap.py in <module>()
----> 1 image_estimator.run(obs_list)

~/Development/gammapy/gammapy/image/basic.py in run(self, observations, which)
    296 
    297             if 'background' in which:
--> 298                 background = self._background(counts, exposure, observation)['background']
    299 
    300                 # TODO: include stacked alpha and on/off exposure images

~/Development/gammapy/gammapy/image/basic.py in _background(self, counts, exposure, observation)
    256         input_images['exposure_on'] = exposure_on
    257         input_images['exclusion'] = self._cutout_observation(self.exclusion_mask, observation)
--> 258         return self.background_estimator.run(input_images)
    259 
    260     def run(self, observations, which='all'):

TypeError: run() takes 1 positional argument but 2 were given
```

**Comment by registerrier** [comment]
Reflected background is indeed not supposed to be used for image analysis, but only for 1D spectral analysis. This will require better documentation and a proper test of background type to avoid this issue.

**Comment by cdeil** [comment]
> This will require better documentation and a proper test of background type to avoid this issue.

I'm very sceptical of the current scheme the `IACTBasicImageEstimator` attempts, to take a background estimator class as input. The motivation is of course to allow for different ways to make images (built-in and even user-defined), and to avoid code duplication with the other part one usually wants, namely make a counts and exposure image as well as a PSF kernel.

@registerrier - I think in the new scheme that you will propose soon, this bug of someone passing a "background estimator" to an "image estimator" that isn't compatible doesn't exist any more, right?
My current understanding of the new scheme: the equivalent of `IACTBasicImageEstimator` would be a function or class that contains a common pre-scripted recipe of steps. And it's not done in an extensible, composable way, but with if / else there's a few pre-scripted options available only. And if a user wants to do something more custom, using their own background estimation or chaining available algorithms in Gammapy in a different way, they would have to either script the lower-level functions together directly, or copy & paste & edit the existing examples how to run an analysis pipeline.

So basically Gammapy as a **library** of algorithms one can compose, very flexible. And Gammapy is not a **framework** that one can extend and plug in by subclassing or generally passing in functions and classes that are then called back from the Gammapy framework at the appropriate step in an analysis.

So basically my suggestion here is to just deprecate the existing code soon (once the new one is in place and documented), no other action.

@adonath might have an opinion or some insight on this very important question as well.
